app/models/service_template_provision_task.rb

```diff
@@ -9,6 +9,10 @@ def self.base_model
     ServiceTemplateProvisionTask
   end
 
+  def my_zone
+    options.fetch_path(:dialog, "dialog_zone") || source.my_zone || MiqServer.my_zone
```

Comment:
You want the same `dialog_zone` method for the task as it has the same exposure.